languages/python/sqlalchemy-oso/sqlalchemy_oso/roles2.py

```diff
@@ -53,7 +58,7 @@ def role_allow_query(
         with resources as (
             with recursive resources (id, type) as (
                 select
-                    {resource_id_field} as id,
+                    cast({resource_id_field} as {resource_id_type}) as id,
```

Comment:
weren't we trying to avoid `cast()`s b/c they'd skip indices?

Reply:
Casts were bad when we were casting the id column on a resource to be a string (if it was an int or something else it'd have to table scan instead of use the index).
In this case we're just casting the input argument to the query to match the id column type, so it's good.
This was failing in some cases when the arg came in as a `text` and the column was a `varchar`. Sqlite didn't care but postgres is stricter.

Reply:
classic pg